### DISCUSSION TITLE
Logical Logging

### DISCUSSION THREAD

**Discussion by Heulitig** [fastn-stack/fastn#1801] [Ideas & RFCs]
## Logical logging on fastn auth and ft workers 

- [x] Basic ActionKind, EntityKind, Outcome, LogLevel implementation 
- [x] Logs to Event
- [x] Removed user-agent from headers and password logging from req body
- [x] Logging: Auth routes
  - [x] Login
    - [x] Info
    - [x] Error
    - [x] Success
  - [x] Github Login 
    - [x] Info
    - [x] Error
    - [x] Success
  - [x] GithubCallback
    - [x] Info
    - [x] Error
    - [x] Success
  - [x] CreateAccount
    - [x] Info
    - [x] Error
    - [x] Success
  - [x] EmailConfirmationSent
    - [x] Info
    - [x] Error
    - [x] Success
  - [x] ConfirmEmail
    - [x] Info
    - [x] Error
    - [x] Success
  - [x] ResendConfirmEmail
    - [x] Info
    - [x] Error
    - [x] Success
  - [x] Onboarding
    - [x] Info
    - [x] Error
    - [x] Success
  - [x] ForgotPassword
    - [x] Info
    - [x] Error
    - [x] Success
  - [x] ForgotPasswordSuccess
    - [x] Info
    - [x] Error
    - [x] Success
  - [x] Logout
    - [x] Info
    - [x] Error
    - [x] Success
  - [x] SetPassword
    - [x] Info
    - [x] Error
    - [x] Success
  - [x] SetPasswordSuccess
    - [x] Info
    - [x] Error
    - [x] Success
  - [x] InvalidRoute
    - [x] Error
- [x] Group Descriptive Error Outcomes - server-error, form-error, unauthorized-error
- [ ] Fixes:
  - [x] Remove False Success logging for FormErrors (user-error) which are returned as 200(OK) response 
  - [ ] Response time and other metric fixes
  - [ ] Action/Views outcome_data to JSON
  - [ ] Code cleanups (if any)
- [ ] Log Testing
- [ ] Logging: Workers

**Comment by amitu** [comment]
@Heulitig for every logical log you add, do a SQL query after doing that event, and show the event data in database. I want to see everything that is getting logged in the table.

**Reply by Heulitig** [reply]
I did this SQL query to check the login logs

Query: `select id, ekind, okind, outcome, message, source, data from ft_event order by id desc limit 5;`

[
<img width="1393" alt="Screenshot 2024-03-06 at 1 40 58 PM" src="https://github.com/fastn-stack/fastn/assets/106665190/4d9f5759-3d8b-4a96-ba3c-b708c408fd53">
](url)

I will improve the outcome for errors currently it shows error for all kinds of errors

**Reply by amitu** [reply]
No, do a select *, do not hide information. You will see the columns you have hidden have wrong values.

**Reply by amitu** [reply]
Make sure you do not log the password. Also delete user-agent header as you have a dedicated column for that.

**Comment by amitu** [comment]
We are creating a [new discussion for this](https://github.com/orgs/fastn-stack/discussions/2159).
